src/System.Private.CoreLib/shared/System/Globalization/TextInfo.cs

```diff
@@ -909,5 +1009,11 @@ private static bool IsLetterCategory(UnicodeCategory uc)
                  || uc == UnicodeCategory.ModifierLetter
                  || uc == UnicodeCategory.OtherLetter);
         }
+
+        // A dummy struct that is used for 'ToUpper' in generic parameters
+        private readonly struct ToUpperConversion { }
```

Comment:
You can also do this by using aliases for existing types:
```
using ToUpperConversion = System.UInt32;
using ToLowerConversion = System.Int32;
```
It is a micro-optimization. Feel free to ignore this if you think explicit types are better.